src/Voron/Impl/Journal/JournalWriter.cs

```diff
@@ -28,49 +28,87 @@ public unsafe class JournalWriter : IJournalWriter
             if (result != PalFlags.FailCodes.Success)
                 PalHelper.ThrowLastError(result, error, $"Attempted to open journal file - Path: {filename.FullPath} Size :{size}");
 
             NumberOfAllocated4Kb = (int)(actualSize / (4 * Constants.Size.Kilobyte));
         }
 
-        public TimeSpan Write(long posBy4Kb, byte* p, int numberOf4Kb)
+        public TimeSpan Write(long posBy4Kb, byte* buffer, int numberOf4Kb)
         {
             Debug.Assert(_options.IoMetrics != null);
 
             Stopwatch sp;
-
             using (var metrics = _options.IoMetrics.MeterIoRate(FileName.FullPath, IoMetrics.MeterType.JournalWrite, numberOf4Kb * 4L * Constants.Size.Kilobyte))
             {
                 sp = Stopwatch.StartNew();
 
-                var result = Pal.rvn_write_journal(_writeHandle, p, numberOf4Kb * 4L * Constants.Size.Kilobyte, posBy4Kb * 4L * Constants.Size.Kilobyte, out var error);
-                if (result != PalFlags.FailCodes.Success)
-                    PalHelper.ThrowLastError(result, error, $"Attempted to write to journal file - Path: {FileName.FullPath} Size: {numberOf4Kb * 4L * Constants.Size.Kilobyte}, numberOf4Kb={numberOf4Kb}");
+                // We will separate them in chunks of max size. 
+                long iterations = numberOf4Kb / _maxNumberOf4KbInSingleWrite;
+                long leftovers = numberOf4Kb % _maxNumberOf4KbInSingleWrite;
 
-                sp.Stop();
+                byte* currentBufferPtr = buffer;
+                for (long i = 0; i < iterations; i++ )
+                {
+                    using (_writeSynchronization.Enter())
+                    {
+                        var result = Pal.rvn_write_journal(_writeHandle, currentBufferPtr, _maxNumberOf4KbInSingleWrite * 4L * Constants.Size.Kilobyte, posBy4Kb * 4L * Constants.Size.Kilobyte, out var error);
+                        if (result != PalFlags.FailCodes.Success)
+                            PalHelper.ThrowLastError(result, error, $"Attempted to write to journal file - Path: {FileName.FullPath} Size: {numberOf4Kb * 4L * Constants.Size.Kilobyte}, numberOf4Kb={numberOf4Kb}");
+
+                        if (error == ERROR_WORKING_SET_QUOTA && _log.IsOperationsEnabled && _workingSetQuotaLogged == false)
+                        {
+                            _log.Operations(
+                                $"We managed to accomplish journal write although we got {nameof(ERROR_WORKING_SET_QUOTA)} under the covers and wrote data in 4KB chunks");
+
+                            _workingSetQuotaLogged = true;
+                        }
+                    }
+
+                    posBy4Kb += _maxNumberOf4KbInSingleWrite;
+                    currentBufferPtr += _maxNumberOf4KbInSingleWrite * 4L * Constants.Size.Kilobyte;
+                }
 
-                if (error == ERROR_WORKING_SET_QUOTA && _log.IsOperationsEnabled && _workingSetQuotaLogged == false)
+                // Then we will write the last chunk. 
```

Comment:
We can remove this.

Reply:
I suffered a bit after coming back the lack of comments, agree this particular one is not so important, but I will be writing a lot more from now on.

Reply:
That isn't about the comments, that is about handling the last portion, there isn't going to be one.
It is always 4KB aligned.